lib/rspec/core/notifications.rb

```diff
@@ -496,7 +496,8 @@ def duplicate_rerun_locations
     # @attr duration [Float] the time taken (in seconds) to run the suite
     # @attr examples [Array<RSpec::Core::Example>] the examples run
     # @attr number_of_examples [Fixnum] the number of examples to profile
-    ProfileNotification = Struct.new(:duration, :examples, :number_of_examples)
+    # @attr profiler [RSpec::Core::Profiler] the profiler used to capture examples
+    ProfileNotification = Struct.new(:duration, :examples, :number_of_examples, :profiler)
```

Comment:
> Why would a formatter have to gain access to another formatter via a notification?

It's not a formatter, it's the profiler, `RSpec::Core::Profiler` not `RSpec::Core::ProfileFormatter`. This is *given* to `RSpec::Core::ProfileFormatter`. One way, no circular dependency.